src/data/markdown/translated-guides/en/01 Getting started/02 Installation/01 Troubleshooting.md

```diff
@@ -10,22 +10,49 @@ If you use such a distribution, you'll need to install them:
 
 ```bash
 sudo apt-get update && sudo apt-get install ca-certificates gnupg2 -y
-sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys C5AD17C747E3415A3642D57D77C6C491D6AC1D69
-echo "deb https://dl.k6.io/deb stable main" | sudo tee /etc/apt/sources.list.d/k6.list
+sudo gpg --no-default-keyring --keyring /usr/share/keyrings/k6-archive-keyring.gpg --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys C5AD17C747E3415A3642D57D77C6C491D6AC1D69
+echo "deb [signed-by=/usr/share/keyrings/k6-archive-keyring.gpg] https://dl.k6.io/deb stable main" | sudo tee /etc/apt/sources.list.d/k6.list
 sudo apt-get update
 sudo apt-get install k6
 ```
 
+## apt-key is deprecated
+
+Some users might run into an `apt-key` deprecation warning while adding k6's repository key to their system's keyring: `Warning: apt-key is deprecated. Manage keyring files in trusted.gpg.d instead (see apt-key(8))`. Ubuntu and Debian deprecated the `apt-key` command indeed. In order to avoid it, and be future-proof, users should delete the existing `security@k6.io` existing repository key present on their system, and update their sources list accordingly.
+
+```bash
+# delete existing key
+apt-key del k6
+
+# import key, the up to date way
+sudo gpg --no-default-keyring --keyring /usr/share/keyrings/k6-archive-keyring.gpg --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys C5AD17C747E3415A3642D57D77C6C491D6AC1D69
+
+# update /etc/apt-sources.list.d/k6.list
+echo "deb [signed-by=/usr/share/keyrings/k6-archive-keyring.gpg] https://dl.k6.io/deb stable main" | sudo tee /etc/apt/sources.list.d/k6.list
+```
+
+## Dirmng is missing or uninitialized
+
+Some users might encounter an error mentioning `No dirmng` as they try to download k6's GPG key from ubuntu's keyserver:
+```bash
+gpg: keybox '/usr/share/keyrings/k6-archive-keyring.gpg' created
+gpg: failed to create temporary file '/root/.gnupg/.#lk0x000055db689f2310.a86c4b090dc7.7': No such file or directory
+gpg: connecting dirmngr at '/root/.gnupg/S.dirmngr' failed: No such file or directory
+gpg: keyserver receive failed: No dirmngr
+```
+
+If this issue affects you, make sure `dirmngr` is installed on your system by running `apt-get install dirmngr` (or equivalent for your distribution),
+and run `dirmngr -q` once to initialize it. 
```

Comment:
That error is complaining that `/root/.gnupg` doesn't exist, and the `No dirmngr` message is informational. Creation of `$GNUPGHOME` happens on the first `gpg` invocation, and `sudo gpg -k` is enough to do that. So users don't need to install `dirmngr`, which is a "server for managing and downloading certificate revocation lists", and most users don't need it.

Reply:
Thanks a lot for the clarification! I ran into the issue myself but didn't look much further into it. Glad to find out you do know much more about it :) Will incorporate your feedback 🙇🏻

Reply:
Oh, I've been using GnuPG for a long time, but I'm still puzzled by _a lot_ of it. :confused:

Reply:
I made the changes in c086af05 (#646).

Reply:
Perfect! Thanks @imiric 🙇🏻